src/fs/metadata_to_disk_buffer.hh

```diff
@@ -0,0 +1,249 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+#pragma once
+
+#include "bitwise.hh"
+#include "metadata_disk_entries.hh"
+#include "to_disk_buffer.hh"
+
+#include <boost/crc.hpp>
+
+namespace seastar::fs {
+
+class metadata_to_disk_buffer : protected to_disk_buffer {
+    boost::crc_32_type _crc;
+
+public:
+    // Represents buffer that will be written to a block_device at offset @p disk_alligned_write_offset. Total number of bytes appended cannot exceed @p aligned_max_size.
+    metadata_to_disk_buffer(size_t aligned_max_size, unit_size_t alignment, disk_offset_t disk_aligned_write_offset)
+    : to_disk_buffer(aligned_max_size, alignment, disk_aligned_write_offset) {
+        assert(aligned_max_size >= sizeof(ondisk_type) + sizeof(ondisk_checkpoint));
+    }
+
+    using to_disk_buffer::reset;
+
+    /**
+     * @brief Clears buffer, leaving it in state as if it was just constructed
```

Comment:
I think that it doesn't leave metadata_to_disk_buffer in any state that is reachable directly after constructing metadata_to_disk_buffer. In that case, that comment is misleading.

Reply:
I changed the comment to be more appropriate.